src/partitioning/visitor.rs

```diff
@@ -78,6 +78,39 @@ where
         (self)(bv, data)
     }
 }
+
+/// Trait implemented by visitor called during the traversal of a spatial partitioning data structure.
+pub trait SimdVisitorContext<LeafData, SimdBV, Context: Clone> {
+    /// Execute an operation on the content of a node of the spatial partitioning structure.
+    ///
+    /// Returns whether the traversal should continue on the node's children, if it should not continue
+    /// on those children, or if the whole traversal should be exited early. Also returns
+    /// a context, which may or may not be identical to the input context.
+    fn visit(
+        &mut self,
+        bv: &SimdBV,
+        data: Option<[Option<&LeafData>; SIMD_WIDTH]>,
+        context: Context,
+    ) -> (SimdVisitStatus, Context);
+}
+
+impl<F, LeafData, SimdBV, Context: Clone> SimdVisitorContext<LeafData, SimdBV, Context> for F
+where
+    F: FnMut(
+        &SimdBV,
+        Option<[Option<&LeafData>; SIMD_WIDTH]>,
+        Context,
+    ) -> (SimdVisitStatus, Context),
+{
```

Comment:
If we can only one blanket impl, I’d find the following one more useful:
```rust
impl<T, LeafData, SimdBV, Context> SimdVisitorWithContext<LeafData, SimdBV, Context> for T
where 
    T: SimdVisitor<LeafData, SimdBV>
{ ... }
```
that way we can easily give a non-contextual visitor to a generic function that allows contexts.

(In a ideal world, I would love if both your blanked impl and the one I suggested coexisted.)